frontend/public/components/hpa.tsx

```diff
@@ -34,64 +38,61 @@ const MetricsRow: React.FC<MetricsRowProps> = ({ type, current, target }) => (
 
 const externalRow = (metric, current, key) => {
   const { external } = metric;
-  const type = external.metricName;
-  // TODO: show metric selector for external metrics?
-  const currentValue = external.targetAverageValue
-    ? _.get(current, 'object.currentAverageValue')
-    : _.get(current, 'object.currentValue');
-  const targetValue = external.targetAverageValue
-    ? external.targetAverageValue
-    : external.targetValue;
+  const type = external.metric.name;
+  const currentValue =
+    current?.external?.current?.averageValue || current?.external?.current?.value;
+  const targetValue = external.target.averageValue || external.target.value;
 
   return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
 };
 
-const objectRow = (metric, current, ns, key) => {
+const objectRow = (metric, current, ns, key, kind) => {
   const { object } = metric;
   const type = (
     <>
-      {object.metricName} on
+      {object?.metric?.name} on
       <ResourceLink
-        kind={object.target.kind}
-        name={object.target.name}
+        kind={kind}
+        name={object?.metric?.name}
```

Comment:
This doesn't look right. I think this should be the scale target, not the name of the metric.

You should set up the HPA to target a real resource and make sure the link works.

Reply:
@spadgett Addressed, thanks